.golangci.yml

```diff
@@ -12,371 +12,135 @@
-    # Enable to require an explanation of nonzero length after each nolint directive. Default is false.
-    require-explanation: true
-    # Enable to require nolint directives to mention the specific linter being suppressed. Default is false.
-    require-specific: true
-  rowserrcheck:
-    packages:
-      - github.com/jmoiron/sqlx
-  testpackage:
-    # regexp pattern to skip files
-    skip-regexp: (export|internal)_test\.go
-  unparam:
-    # Inspect exported functions, default is false. Set to true if no external program/library imports your code.
-    # XXX: if you enable this setting, unparam will report a lot of false-positives in text editors:
-    # if it's called for subdir of a project it can't find external interfaces. All text editor integrations
-    # with golangci-lint call it on a directory with the changed file.
-    check-exported: false
-  unused:
-    # treat code as a program (not a library) and report unused exported identifiers; default is false.
-    # XXX: if you enable this setting, unused will report a lot of false-positives in text editors:
-    # if it's called for subdir of a project it can't find funcs usages. All text editor integrations
-    # with golangci-lint call it on a directory with the changed file.
-    check-exported: false
-  whitespace:
-    multi-if: false   # Enforces newlines (or comments) after every multi-line if statement
-    multi-func: false # Enforces newlines (or comments) after every multi-line function signature
-  wsl:
-    # If true append is only allowed to be cuddled if appending value is
-    # matching variables, fields or types on line above. Default is true.
-    strict-append: true
-    # Allow calls and assignments to be cuddled as long as the lines have any
-    # matching variables, fields or types. Default is true.
-    allow-assign-and-call: true
-    # Allow multiline assignments to be cuddled. Default is true.
-    allow-multiline-assign: true
-    # Allow declarations (var) to be cuddled.
-    allow-cuddle-declarations: false
-    # Allow trailing comments in ending of blocks
-    allow-trailing-comment: false
-    # Force newlines in end of case at this limit (0 = never).
-    force-case-trailing-whitespace: 0
-    # Force cuddling of err checks with err var assignment
-    force-err-cuddling: false
-    # Allow leading comments to be separated with empty liens
-    allow-separated-leading-comment: false
-  gofumpt:
-    # Choose whether or not to use the extra rules that are disabled
-    # by default
-    extra-rules: false
-
+  revive:
+    rules:
+      # Using += 1 instead of ++ is fine
+      - name: increment-decrement
+        disabled: true
+  stylecheck:
+    # Dot importing ginkgo and gomega is standard practice
+    dot-import-whitelist:
+      - "github.com/onsi/gomega"
+      - "github.com/onsi/ginkgo/v2"
 linters:
   enable:
-    - megacheck
-    - govet
-  disable:
-    - maligned
-    - prealloc
-    - scopelint
+    - asasalint
```

Comment:
Probably better to use `enable-all` (and the list of disables below) rather than listing every enable _and_ disable?

Reply:
I thought about this but then every time golang-ci updates we risk blocking CI because of new linters being added. 

Also there are some warnings because deprecated linters are still enabled in enable-all. 

IMO better to explicitly opt in to avoid this annoyance. I see no significant drawback to being more explicit about the linters we allow, we can always add more linters later.

Reply:
> The primary drawback is that we have to search for and explicitly enable any new linter

Yep totally agree that we will miss out on new linters until someone explicitly adds them. (Though is that a bad thing)

I thought about adding a dependency upgrades runbook step of "If upgrading golang-ci, run with `--enable-all` to spot any new linters" but that is even more annoying than upgrading + manually disabling...

> deprecated linters is unfortunate

Yeah I couldn't find anything either. I'll consider raising an issue? But to be fair deprecated linter warnings are a small annoyance. 

---

Thanks for your thoughts, I don't have a strong opinion either way, it's a two way door that boils down to: 

"Do we want the person doing dependency upgrades to enable/disable new linters as they are added, or should we batch those decisions every once in a while"

I'll default to enabling all linters as you originally suggested unless @ElijahQuinones or @torredil have a preference on ^^